spec/requests/apartments_spec.rb

```diff
@@ -24,8 +24,42 @@
       apartment = JSON.parse(response.body)
       expect(response).to have_http_status(200)
       expect(apartment.length).to eq 1
+    end
+  end
 
+  describe "POST /create" do
+    it "creates a apartment" do
+      # The params we are going to send with the request
+      user1 = User.where(email: "test1@example.com").first_or_create(password: "password", password_confirmation: "password")
 
+      apartment_params = {
+      apartment: {
+          street: 'ABC Sesame Street',
+          unit: '20',
+          city: 'Sesame',
+          state: 'MUPPET',
+          square_footage: 3000,
+          price: "1900",
+          bedrooms: 5,
+          bathrooms: 3,
+          pets: "puppets only",
+          user_id: user1.id,
+          image: "https://upload.wikimedia.org/wikipedia/commons/0/00/Sesame_Street_buildings_%28193090534%29.jpg"
+      }
+      }
+
+  
+      # Send the request to the server
+      post '/apartments', params: apartment_params
+  
+      # Assure that we get a success back
+      expect(response).to have_http_status(200)
+  
+      # Look up the cat we expect to be created in the db
+      apartment = Apartment.all
+
+      # Assure that the created cat has the correct attributes
```

Comment:
Some of your pseudocode looks a bit off 😹

Reply:
hahaha shhhhhh